src/blockmap.jl

```diff
@@ -141,51 +149,58 @@ function Base.hvcat(rows::NTuple{nr,Int}, As::Union{LinearMap,UniformScaling}...
         end
     end
 
-    return BlockMap{T}(promote_to_lmaps(n, 1, As...), rows)
+    return BlockMap{T}(promote_to_lmaps(n, 1, 1, As...), rows)
 end
 
-promote_to_lmaps_(n::Int, J::UniformScaling) = UniformScalingMap(J.λ, n)
-promote_to_lmaps_(n::Int, A::LinearMap) = A
-promote_to_lmaps(n, k) = ()
-promote_to_lmaps(n, k, A) = (promote_to_lmaps_(n[k], A),)
-promote_to_lmaps(n, k, A, B) = (promote_to_lmaps_(n[k], A), promote_to_lmaps_(n[k+1], B))
-promote_to_lmaps(n, k, A, B, C) =
-    (promote_to_lmaps_(n[k], A), promote_to_lmaps_(n[k+1], B), promote_to_lmaps_(n[k+2], C))
-promote_to_lmaps(n, k, A, B, Cs...) =
-    (promote_to_lmaps_(n[k], A), promote_to_lmaps_(n[k+1], B), promote_to_lmaps(n, k+2, Cs...)...)
+promote_to_lmaps_(n::Int, dim, J::UniformScaling) = UniformScalingMap(J.λ, n)
+promote_to_lmaps_(n::Int, dim, A::LinearMap) = (check_dim(A, dim, n); A)
+promote_to_lmaps(n, k, dim) = ()
+promote_to_lmaps(n, k, dim, A) = (promote_to_lmaps_(n[k], dim, A),)
+promote_to_lmaps(n, k, dim, A, B) = (promote_to_lmaps_(n[k], dim, A), promote_to_lmaps_(n[k+1], dim, B))
+promote_to_lmaps(n, k, dim, A, B, C) =
+    (promote_to_lmaps_(n[k], dim, A), promote_to_lmaps_(n[k+1], dim, B), promote_to_lmaps_(n[k+2], dim, C))
+promote_to_lmaps(n, k, dim, A, B, Cs...) =
+    (promote_to_lmaps_(n[k], dim, A), promote_to_lmaps_(n[k+1], dim, B), promote_to_lmaps(n, k+2, dim, Cs...)...)
```

Comment:
This last method also captures the previous two, no? Adding `@inline` in front should make sure that there is no performance difference with or without having the former two.

Reply:
Yes, I commented out the previous two lines and `@inline`d the `A, B, Cs...` one, and it seems to work fine with one, two, three and more linear maps. So I removed the previous two lines.

Reply:
Oh, and I guess that change will actually help us test the method with no linear map argument. 🎉